lib/req_ch.ex

```diff
@@ -127,6 +127,59 @@ defmodule ReqCH do
     |> Req.post(body: sql_query)
   end
 
+  @doc """
+  Performs a query against the ClickHouse API.
+
+  This version receives a `Req.Request.t()`, so it won't
+  create a new one from scratch.
+
+  By default, it will use the `http://localhost:8123` as `:base_url`.
+  You can change that either providing in your Req request, or in passing
+  down in the options.
+  See `new/1` for the options.
+
+  ## Examples
+
+  A plain query:
+
+      iex> req = ReqCH.new(database: "system")
+      iex> {:ok, response} = ReqCH.query(req, "SELECT number FROM numbers LIMIT 3", [], [])
+      iex> response.body
+      "0\\n1\\n2\\n"
+
+  With a specific format:
+
+      iex> req = ReqCH.new(database: "system")
+      iex> {:ok, response} = ReqCH.query(req, "SELECT number FROM numbers LIMIT 3", [], [format: :explorer])
+      iex> response.body
+      #Explorer.DataFrame<
+        Polars[3 x 1]
+        number u64 [0, 1, 2]
+      >
+
+   Passing SQL params:
+
+      iex> req = ReqCH.new(database: "system")
+      iex> {:ok, response} = ReqCH.query(req, "SELECT number FROM numbers WHERE number > {num:UInt8} LIMIT 3", [num: 5], [])
+      iex> response.body
+      "6\\n7\\n8\\n"
+  """
+  @spec query(
+          Req.t(),
+          sql_query :: binary(),
+          sql_query_params :: Map.t() | Keyword.t(),
+          opts :: Keyword.t()
+        ) :: {:ok, Req.Response.t()} | {:error, binary()}
+  def query(req, sql_query, sql_query_params, opts)
```

Comment:
Once you delete `query/3`:

```suggestion
  def query(req, sql_query, sql_query_params \\ [], opts \\ [])
```